cecrets/cecrets.go

```diff
@@ -89,11 +91,17 @@ type secret struct {
 }
 
 // use the hashed string in any fmt verb.
-func (s secret) Format(fs fmt.State, verb rune) { io.WriteString(fs, s.hashText) }
-func (s secret) String() string                 { return s.hashText }
-func (s secret) Conceal() string                { return s.hashText }
-func (s secret) PlainString() string            { return s.plainText }
-func (s secret) V() any                         { return s.value }
+func (s secret) Format(fs fmt.State, verb rune) {
+	_, err := io.WriteString(fs, s.hashText)
+	if err != nil {
+		log.Printf("error writing string: %+v\n", err)
```

Comment:
sanity check: does writing to the log feel like the best way to handle this? I know some logging systems ignore errors. Having uncontrolled output from a library can also sometimes make it difficult to use the library

If we need to allow tuning the output, maybe a config flag for "mode" could be added so that callers could differentiate between stuff like "prod" and "dev"?

Reply:
That output config is definitely something planned for the future.  In the current time I've been switching these from io.Writers to fmt.Fprinters to avoid the error linting issue.  It's a punt, but should work for the cases here.  And is a two-way door should we need to backtrack.

Reply:
That output config is definitely something planned for the future.  In the current time I've been switching these from io.Writers to fmt.Fprinters to avoid the error linting issue.  It's a punt, but should work for the cases here.  And is a two-way door should we need to backtrack.

Reply:
well, that will take care of most of the cases, at least.